### ISSUE TITLE
Add a release pipeline

### ISSUE THREAD

**Issue by iamandylamb** [iamandylamb/ExplicitMapper#14]
I'd like to have:
* An automated publish to nuget.org.
Not on every PR, but when a release is ready. Not sure what we'd use to trigger it though?

**Comment by iamandylamb** [comment]
This might be useful:
https://github.com/marketplace/actions/publish-nuget

**Comment by colin-gourlay** [comment]
The [Nuget Action](https://github.com/marketplace/actions/publish-nuget) does look useful.

Have you any thoughts on the flow that you want to adopt? I really like the simplicity of the [GitHub Flow](https://guides.github.com/introduction/flow/) but if you are looking to release only when it is ready, then might a variant of the Git Flow be 'better/easier' to manage using GitHub Actions? I think introducing a `release` branch and associating the Action with that might do the job.

Something like this:

<pre>

release   |-----------------------------0-------------------------------->
                                       / 
master    |-0---------0-----0---------0---------------------------------->
             \       /       \       /
pr's      |---0->0->0---------0->0->0------------------------------------>

   
</pre>

Any thoughts?

**Comment by feelthesource** [comment]
Yes, it's a good point. I think Git flow is probably the best bet. But let's keep it 'by the book'. Let's put a trigger on `master` to release to NuGet and use a `develop` branch to integrate pull requests before merging to `master`.

**Comment by iamandylamb** [comment]
I've added the `develop` branch and set up some branch rules for `master` and `develop`. So should be good to go with setting up the release action.
